### PR TITLE
[Time Conductor] add history and select range features  to vista-r4.4.1

### PR DESCRIPTION
**Pull request by davetsay**
## Overview
Addresses  #2430, Supercedes #2552

- change pan axis to require alt key press and hold
- add zoom axis
  - drag to select a range within current bounds to "zoom" in to
  - zoomed range becomes new bounds
- visually highlight  on pan and on zoom
- time conductor history
  - records recently used times based on current time system
  - saves time history on bounds change
  - saves time history on time system change
  - configurable presets (ex. last hour, last day)
  - configurable number of previous history records
  - persists history for all used time systems in local storage

## Author Checklist
| | |
| --- | :---: |
| Changes address original issue? | Y |
| Unit tests included and/or updated with changes? | N/A |
| Command line build passes? | Y |
| Changes have been smoke-tested? | Y |

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -63,15 +63,40 @@
...
+                        {
+                            label: 'Last hour',
+                            bounds: {
+                                start: Date.now() - 1000 * 60 * 60,
+                                end: Date.now()
+                            }
+                        }
+                    ],
+                    // maximum entries to retain in conductor history
+                    records: 10
```
Comment (deeptailor) [inline index.html:91]: Can we have the records preset be set as default, so when timesystems dont have this option set, it uses the default 10 vs failing completely

File: index.html [hunk 2]
```diff
@@ -88,7 +88,9 @@
                         }
                     ],
                     // maximum entries to retain in conductor history
-                    records: 10
+                    records: 10,
+                    // maximum duration between start and end bounds
```
Comment (deeptailor) [inline index.html:92]: lets add a clear comment that this limit is in milliseconds

File: src/plugins/timeConductor/Conductor.vue [hunk 3]
```diff
@@ -175,29 +194,64 @@ export default {
...
+            isZooming: false
         }
     },
+    created() {
+        document.addEventListener('keydown', (e) => {
+            if (e.key === 'Alt') {
+                this.altPressed = true;
+            }
+        });
+        document.addEventListener('keyup', (e) => {
```
Comment (deeptailor) [inline src/plugins/timeConductor/Conductor.vue:215]: lets clean up the event listeners on destroy. I know that the time conductor is only going to be created and destroyed once with the life of the application, but its good practice, and promotes reusability of this component in the future.

File: index.html (hunk 1 above)
Comment (davetsay) [reply index.html]: Yes. See this line in ConductorHistory.vue
`const records = configuration ? configuration.records : DEFAULT_RECORDS;`

File: src/plugins/timeConductor/Conductor.vue (hunk 3 above)
Comment (davetsay) [reply src/plugins/timeConductor/Conductor.vue]: ah yes. thanks. done.

File: src/plugins/timeConductor/Conductor.vue [hunk 4]
```diff
@@ -175,29 +194,68 @@ export default {
...
+            this.isPanning = true;
+            this.setViewFromBounds(bounds);
+        },
+        endPan(bounds) {
+            this.isPanning = false;
+            if (bounds) {
+                this.openmct.time.bounds(bounds);
+            }
+        },
+        zoom(bounds) {
```
Comment (deeptailor) [inline src/plugins/timeConductor/Conductor.vue:243]: lets name this startZoom, since we have an endZoom

File: index.html (hunk 2 above)
Comment (davetsay) [reply index.html:96]: done

File: src/plugins/timeConductor/Conductor.vue (hunk 4 above)
Comment (davetsay) [reply src/plugins/timeConductor/Conductor.vue:243]: We have startZoom (and zoom and endZoom).

Comment (davetsay) [reply src/plugins/timeConductor/Conductor.vue:243]: oh i see. nevermind. this is in conductor.vue.

zoom in conductor is called on both startZoom and zoom in conductor axis. I could have two events to mimic startZoom and zoom but i thought this was cleaner.

Comment (deeptailor) [reply src/plugins/timeConductor/Conductor.vue:243]: ok, no problem. I think it is ok as it stands

File: src/plugins/timeConductor/ConductorHistory.vue [hunk 5]
```diff
@@ -160,11 +160,13 @@ export default {
         selectTimespan(timespan) {
             this.openmct.time.bounds(timespan);
         },
-        selectHours(hours) {
-            const now = Date.now();
+        selectPresetBounds(bounds) {
+            const start = typeof bounds.start === 'function' ? bounds.start() : bounds.start;
```
Comment (deeptailor) [inline src/plugins/timeConductor/ConductorHistory.vue:164]: Looks good

**Comment by deeptailor** [comment]
## Reviewer Checklist
Changes appear to address issue? Yes
Appropriate unit tests included? N/A
Code style and in-line documentation are appropriate? Yes
Commit messages meet standards? Yes

**Pull request merged**